### ISSUE TITLE
Do not add commands to shell history

### ISSUE THREAD

**Issue by Kriechi** [greymd/tmux-xpanes#89]
Hi,

I'm using macOS, with zsh (and oh-my-zsh) from homebrew in iTerm.
My history behaves in such a way that it is shared across running shells. This means a new history item is available to the up/down auto-completion across all my running windows/panes/tmux sessions/terminal windows/etc.

When I run `xpanes -c 'ssh server-{}' {1..10}`, and exit all ssh sessions, I can see that xpanes actually added ten new lines to my `.zsh_history` history file. 

I did not expect that, and would prefer it to only add the actual xpanes invocation to the history.

Edit:
I guess this is because xpanes uses `send-keys` (https://github.com/greymd/tmux-xpanes/blob/3fcdbfdaf0c0cb22cbed92e7d3208b79fb60a1ed/bin/xpanes#L503) instead of invoking tmux new-pane with a shell command directly.

**Comment by greymd** [comment]
Hi @Kriechi,

This is also what I want to solve!
hmm..., but I do not have any good ideas for it for now.
Users may use not only zsh, but also bash, fish, tcsh and so on.
Unfortunately, the way to suppress the logging of shell history is depending on the shell.
In addition, tmux-xpanes cannot detect what shell is executed after sending commands with `send-keys`.

Please let me think how to solve it for a while.

**Comment by Kriechi** [comment]
I think a shell-independet solution should be possible.
I'm wondering why you are using send-keys in the first place?

For example, tmux-cssh uses split-window and directly executes the command:
https://github.com/peikk0/tmux-cssh/blob/master/tmux-cssh#L64

**Comment by greymd** [comment]
The reason why `xpanes` uses `send-keys` is, to keep panes alive and interactive.
xpanes once opens new default login shell and send commands on them.

It seems that, `split-window "command"` creates new pane and execute the command 
 as a direct child process of tmux (not shell). This is why `tmux-cssh` does not make any history.
`ssh` command allocates pseudo-tty and provides interactive screen itself. That's why `tmux-cssh` can keep interactive screen on the separated panes.

But xpanes supports not only ssh but also any commands the user want to run.
If `send-keys` is replaced with `split-window` in `xpanes`, it works but newly created panes will be closed immediately when the process of the command is finished.

But it is possible to create new option to let user choose `split-window` mode (pane is closed immediately when process exits) or `send-keys` mode (keep interactive screen for anytime).
Is it what you want?

**Comment by Kriechi** [comment]
I think tmux also has an option to keep panes open after the command/shell has exited: `remain-on-exit`
Maybe it would be good idea to use this, instead of interactive login shells?

I'm also happy, if you just provide an option to "use split-window instead of send-keys".

**Comment by greymd** [comment]
wow, I did not know `remain-on-exit`.
I will try, but it may take much time.
stay tuned.

**Comment by greymd** [comment]
MEMO:
I am considering to make two new option for the next tmux-xpanes version (v3.1.0, hopefully).

```
-s: option:  Speedy mode. Just execute command. Do not make panes interactive.
-a: keep alive option (do not close pane when the command finished when -s option)
```

**Comment by Kriechi** [comment]
Yes - this sounds perfect for my use cases!

**Comment by greymd** [comment]
This feature is released at [v3.1.0](https://github.com/greymd/tmux-xpanes/releases/tag/v3.1.0)
Try new options `-s` and `-ss` are released. `-a` is reserved for the future release. Here is [reason](https://github.com/greymd/tmux-xpanes/issues/93#issuecomment-408640238).

Plz try them. If it does not satisfies you, re-open or create new issue. thanks.
